types/src/document.rs

```diff
@@ -146,11 +146,16 @@ impl AnnotatedDoc {
 
     /// All the words contained in this document, dropping structural formatting
     /// like line and page breaks.
-    async fn forms(&self, context: &async_graphql::Context<'_>) -> FieldResult<Vec<AnnotatedForm>> {
+    async fn forms(
+        &self,
+        context: &async_graphql::Context<'_>,
+        start_index: Option<i64>,
+        end_index: Option<i64>,
```

Comment:
I might prefer names like `start` and `end` or `first` and `last` (although that implies inclusive on both ends). Your call though.